### PR TITLE
Update bundled libusb to 1.0.23

### LINKED ISSUE #9: New 1.0.23 release
There is a new release and it'd be good to update this when there is time for that.

One thing it would be interesting is to try to automate the binding generation using `bindgen` so it could reduce the burden of updating and finding new additions for every release.

### PR DESCRIPTION
**Pull request by serpilliere**
Fixes #9 
Bump tgz of libusb-1.0.23 and add it's new APIs (libusb_log_cb_mode, libusb_set_cb, libusb_wrap_sys_device)

### CODE REVIEW

**Comment by a1ien** [comment]
Thanks. LGTM. I update crate in this weekend.

**Comment by serpilliere** [comment]
Thanks to you!
Question: when you say "I update crate", do you talk about `libusb-sys`, or `rusb` as well ?
I must admit that I am interested in the new `libusb_wrap_sys_device` function added in the `1.0.23`, and I would like to know if you are interested by a patch for rusb to support this function ?

**Comment by a1ien** [comment]
`libusb-sys`.
If you send patch to `rusb` i'am happy to review it

**Comment by serpilliere** [comment]
Ok, I will do that. Thanks for this!

File: src/lib.rs [hunk 1]
```diff
@@ -216,6 +221,7 @@ extern "system" {
     pub fn libusb_init(context: *mut *mut libusb_context) -> c_int;
     pub fn libusb_exit(context: *mut libusb_context);
     pub fn libusb_set_debug(context: *mut libusb_context, level: c_int);
+    pub fn libusb_set_cb(context: *mut libusb_context, cb: libusb_log_cb, mode: c_int);
```
Comment (avantgardnerio) [inline src/lib.rs:224]: I was trying to use this function in Rust, and hit my head for a few hours before realizing it isn't in libusb our the resulting `.a` file anywhere? Is there a way to get it to use a callback logger?

Comment (a1ien) [reply src/lib.rs:224]: I think this function must be name `libusb_set_log_cb`

Comment (avantgardnerio) [reply src/lib.rs:224]: Yes, I think you're correct. If I get it to work, I'll file a PR.

Comment (serpilliere) [reply src/lib.rs:224]: Oh!
Sorry for this, my bad.
(I am wondering now how the hell I managed to *change* the name of this function during the  patch.)

**Pull request merged**